resources/views/scripts/dark-theme-script.blade.php

```diff
@@ -0,0 +1,3 @@
+@if((bool) config('ui.dark-mode.enabled', false))
+    <script>document.addEventListener("setThemeMode",(e=>{localStorage.theme=e.detail.theme,toggleTheme()})),document.addEventListener("setOSThemeMode",(e=>{localStorage.removeItem("theme"),toggleTheme()})),document.addEventListener("toggleThemeMode",(e=>toggleTheme()));const toggleTheme=()=>{"dark"===localStorage.theme||!("theme"in localStorage)&&window.matchMedia("(prefers-color-scheme: dark)").matches?document.documentElement.classList.add("dark"):document.documentElement.classList.remove("dark")};toggleTheme();</script>
```

Comment:
3 questions:

1. why cast the config value? wouldn't this be enough: `if (config('ui.dark-mode.enabled', false) === true)`?
2. why is the js on a single line? it makes it hard to read :sadmici:
3. why can't it be in a `dark-mode.js` file and that script tag includes the file instead? I presume it's because it would flash the page in light mode initially? :thinking: e.g. why not the below:

```blade
@if((bool) config('ui.dark-mode.enabled', false))
    <script src="/js/dark-mode.js"></script>
@endif
```

Reply:
@alexbarnsley 
1. yes, you're right 👍🏼 
2. the script is loaded inline in every page, I thought it was better to minifying it. But since we use html minifier from Cloudflare, we can safely keep it in non-minified version.
3. for this kind of script (that it should be executed as soon as possible) I prefer inlining the entire script in the page right after the `<title>` tag. Tbh I haven't tested the file version yet, but this introduces a rendering block until the file has been downloaded. Sure, it's a matter of milliseconds but it's always better to avoid it.